io/src/utils.cpp

```diff
@@ -0,0 +1,44 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "traccc/io/utils.hpp"
+
+// System include(s).
+#include <cstdlib>
+#include <iomanip>
+#include <sstream>
+
+namespace traccc::io {
+
+const std::string& data_directory() {
+
+    // Establish the directory name only once for the application.
+    static const std::string data_dir = [] {
+        // Look for an environment variable specifying the directory name.
+        const char* env_dir = std::getenv("TRACCC_TEST_DATA_DIR");
+        if (env_dir == nullptr) {
+            // If an environment variable is not set, rely on the definition
+            // coming from the build system.
+            return std::string(TRACCC_TEST_DATA_DIR) + "/";
```

Comment:
Hmm, this strikes me as potentially dangerous; should we add a check whether `TRACCC_TEST_DATA_DIR` is actually defined? Otherwise we might end up accidentally read or writing from `/`, which is a little spooky.

Reply:
:thinking: If `TRACCC_TEST_DATA_DIR` is not defined as a macro, the compilation should fail. (In that case `TRACCC_TEST_DATA_DIR` becomes an unknown token/symbol during the compilation.)

My idea was that this macro would **always** be defined as a `PRIVATE` macro for the `traccc::io` library. This is one of the main reasons that I moved this function's implementation into a `.cpp` file. To be able to rely on a library-specific macro. (Without it bleeding out into all the clients.)

With all this being said, the way that file/directory names are handled in the code is super fragile. But at this point I think it's not worth to make it more robust than this...